### PR TITLE
Improve QueryEditor flow

### PR DESCRIPTION
**Pull request by Janpot**
- Add SystemDialogs abstraction to increase uniformity
- add confirmation dialog to remove button
- close the queryeditor dialog on save
- close the page url parameters dialog on save

Closes https://github.com/mui/mui-toolpad/issues/798

### CODE REVIEW

**Review by apedroferreira — APPROVED** [review]
LGTM!

File: packages/toolpad-app/src/utils/useBoolean.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import * as React from 'react';
+
+export default function useBoolean(initialValue: boolean) {
+  const [value, setValue] = React.useState(initialValue);
+  const toggle = React.useCallback(() => setValue((existing) => !!existing), []);
+  const set = React.useCallback(() => setValue(true), []);
```
[the remark this answers is not in the document]
Comment (Janpot) [reply packages/toolpad-app/src/utils/useBoolean.ts]: set/reset is quite common nomenclature in boolean logic, but I have no strong feelings about it, I renamed it to `setTrue`/`setFalse`

[the remark this answers is not in the document]
Comment (apedroferreira) [reply packages/toolpad-app/src/utils/useBoolean.ts]: ok i hadn't heard about it, not a big deal. thanks for the change though!

[the remark this answers is not in the document]
Comment (Janpot) [reply packages/toolpad-app/src/utils/useBoolean.ts]: Maybe it's more common in electronic circuits 🙂

**Pull request merged**